moveit_ros/planning/planning_scene_monitor/src/trajectory_monitor.cpp

```diff
@@ -44,7 +45,16 @@ static const rclcpp::Logger LOGGER = rclcpp::get_logger("moveit_ros.planning_sce
 
 planning_scene_monitor::TrajectoryMonitor::TrajectoryMonitor(const CurrentStateMonitorConstPtr& state_monitor,
                                                              double sampling_frequency)
+  : TrajectoryMonitor(state_monitor, std::make_unique<TrajectoryMonitorMiddlewareHandle>(sampling_frequency),
+                      sampling_frequency)
+{
+}
+
+planning_scene_monitor::TrajectoryMonitor::TrajectoryMonitor(
+    const CurrentStateMonitorConstPtr& state_monitor,
+    std::unique_ptr<TrajectoryMonitor::MiddlewareHandle> middleware_handle, double sampling_frequency)
```

Comment:
Judging from https://github.com/ros-planning/moveit2/pull/570/files#diff-f6042f47b79910249d738bcd543e8424d07fb3f45995bc4d567053d00e222e3aR69

It looks like the middleware handle will have to have an interface to change it's frequency since that's what was happening https://github.com/ros-planning/moveit2/pull/570/files#diff-f6042f47b79910249d738bcd543e8424d07fb3f45995bc4d567053d00e222e3aL111 on every loop

Reply:
Wait, wth???
https://github.com/ros-planning/moveit2/pull/570/files#diff-f6042f47b79910249d738bcd543e8424d07fb3f45995bc4d567053d00e222e3aR71
```
if (sampling_frequency != sampling_frequency_)
    return;  // silently return if nothing changes
```
Doesn't this mean you can never change the frequency? @tylerjw am I reading this wrong?

Reply:
lol, that looks like a bug (based on the comment).

It should probably be `(sampling_frequency == sampling_frequency_)`.  I think this is prof that you will need to add an interface to change the frequency used.